components/atomic/buttons/ButtonControl/ButtonControl.tsx

```diff
@@ -7,10 +7,19 @@ type AuthorizeProps = React.ComponentProps<typeof Authorize>;
 
 const ButtonControl = forwardRef(
   ({ children, iconRotate, actions, allowedActions, ...props }: Props, ref) => {
-    const { icon, size } = props;
+    const { icon, size, closeDropdown, onClick } = props;
+
+    const handleClick = onClick
+      ? closeDropdown
+        ? (e: React.MouseEvent) => {
+            closeDropdown();
+            onClick(e);
+          }
+        : onClick
+      : closeDropdown;
 
     const content = (
-      <Styled.ButtonControl ref={ref} {...props}>
+      <Styled.ButtonControl ref={ref} onClick={handleClick} {...props}>
```

Comment:
Here I added a general `handleClick` so `ButtonControl` can manage the new optional `closeDropdown` prop.